DESIGN.md

```diff
@@ -0,0 +1,41 @@
+# Random Provider Design
+
+The Random Provider offers focussed functionality specifically geared towards the generation of random values for use
+within Terraform configurations. Specifically, the provider resources generate random values at the time they are 
+created and then maintain these values unless the resource inputs are altered.
+
+Below we have a collection of _Goals_ and _Patterns_: they represent the guiding principles applied during the
+development of this provider. Some are in place, others are ongoing processes, others are still just inspirational.
+
+## Goals
+
+* [_Stability over features_](.github/CONTRIBUTING.md)
+* Provide managed resources to generate random values for use in the configuration of Terraform configurations.
```

Comment:
Should we explicitly state whether they are cryptographically safe or not?

Reply:
Good call. Have added an entry.